Automattic/pocket-casts-ios
# End of Year: listening time story design

| 📘 Project: #376 |
|:---:|

Add the design for the "listening time" story.

| iPhone 8 | iPhone 14 | iPhone 14 Pro Max |
| -------- | ---------- | ------------------- |
| ![Simulator Screen Shot - iPhone 8 - 2022-10-27 at 09 09 13](https://user-images.githubusercontent.com/7040243/198280507-afee4820-415b-458d-afa6-5bcd635aa6e6.png) | ![Simulator Screen Shot - iPhone 14 - 2022-10-27 at 09 09 47](https://user-images.githubusercontent.com/7040243/198280585-3cbcc3a7-4fd0-4367-a553-9b8f15fdde06.png) | ![Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-27 at 09 10 13](https://user-images.githubusercontent.com/7040243/198280650-7790412b-70f6-4aea-a351-87b688b8d325.png) |

## Details

* This PR also adds the dynamic background based on pdeCcb-wX-p2
* In order to display some podcasts on this screen, I added logic to retrieve the top 10 podcasts, reverse them, and then display the last 3. This way we can display podcasts that the user also listened to (while saving the top 5 for later)
* Colors needed to be downloaded, I didn't added logic for that in this PR to avoid being too big

## To test

I recommend using these devices to test, given they provide different screen sizes:

* iPhone 8
* iPhone 14
* iPhone 14 Pro Max

1. Enable the `endOfYear` flag in `FeatureFlag.swift`
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. When the prompt appears, tap "View My 2022"
4. Navigate to the second story
4. ✅ Check the design

## Checklist
